sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1337,6 +1377,18 @@ def test_neighbors_metrics(n_samples=20, n_features=3, n_query_pts=2, n_neighbor
             # Haversine distance only accepts 2D data
             feature_sl = slice(None, 2) if metric == "haversine" else slice(None)
 
+            if (
+                metric == "minkowski"
+                and metric_params.get("w") is not None
+                and algorithm == "kd_tree"
+            ):
+                pytest.skip(
+                    "algorithm='kd_tree' is not valid for "
+                    "metric='minkowski' with a weight parameter 'w': "
+                    "try algorithm='ball_tree' "
+                    "or algorithm='brute' instead"
+                )
```

Comment:
This should no longer be needed because we already have the following condition above:

```python
if algorithm == "kd_tree" and (
                metric not in neighbors.KDTree.valid_metrics or w is not None
):
    ...
    continue
```